pkg/actions/tools/nix/path.go

```diff
@@ -0,0 +1,29 @@
+package nix
+
+import (
+	"strings"
+
+	"github.com/rsteube/carapace"
+)
+
+// ActionPaths completes paths
+//
+// A path can be one of:
+//   - a local file path (default.nix)
+//   - an http/https URL (https://releases.nixos.org/../nixexprs.tar.xz
+//   - a channel: specifier (channel:nixos-22.05)
+//   - a local channel (<nixpkgs>)
+func ActionPaths() carapace.Action {
+	return carapace.ActionCallback(func(c carapace.Context) carapace.Action {
+		arg := c.CallbackValue
+		if strings.HasPrefix(arg, "<") {
+			return ActionLocalChannels().Invoke(c).Prefix("<").Suffix(">").ToA()
+		} else if strings.HasPrefix(arg, "http:") || strings.HasPrefix(arg, "https:") {
+			return carapace.ActionValues()
+		} else if strings.HasPrefix(arg, "channel:") {
+			return ActionRemoteChannels().Invoke(c).Prefix("channel:").ToA()
```

Comment:
I hadn't actually tried this and found out that it's actually wrong. It returns the url where it should be just the actual channel name.

Reply:
Ah, I didn't mean that it was wrong for `nix-channel`. I just misunderstood what it was completing and applied it wrong. I'll probably have to factor out the http request and have it return a different format here instead.

Reply:
I'd say we remove it from `ActionRemoteChannels` and just add it as Prefix where needed.

Reply:
I just saw it and was doing it right now :+1:

Reply:
```diff
diff --git a/pkg/actions/tools/nix/channel.go b/pkg/actions/tools/nix/channel.go
index 930da5a9..62c5ffd4 100644
--- a/pkg/actions/tools/nix/channel.go
+++ b/pkg/actions/tools/nix/channel.go
@@ -63,8 +63,6 @@ func ActionRemoteChannels() carapace.Action {
 		for _, result := range response.Data.Result {
 			vals = append(vals, result.Metric.Channel, result.Metric.Status, styleForStatus(result.Metric.Status))
 		}
-		return carapace.ActionCallback(func(c carapace.Context) carapace.Action {
-			return carapace.ActionStyledValuesDescribed(vals...).Invoke(c).Prefix("https://nixos.org/channels/").ToA()
-		})
+		return carapace.ActionStyledValuesDescribed(vals...)
 	})
 }
```